src/validation.cpp

```diff
@@ -4722,8 +4727,9 @@ void LoadExternalBlockFile(const CChainParams& chainparams, FILE* fileIn, FlatFi
                     if (hash != chainparams.GetConsensus().hashGenesisBlock && !LookupBlockIndex(block.hashPrevBlock)) {
                         LogPrint(BCLog::REINDEX, "%s: Out of order block %s, parent %s not known\n", __func__, hash.ToString(),
                                 block.hashPrevBlock.ToString());
-                        if (dbp)
-                            mapBlocksUnknownParent.insert(std::make_pair(block.hashPrevBlock, *dbp));
+                        if (dbp && blocks_with_unknown_parent) {
+                            blocks_with_unknown_parent->emplace(block.hashPrevBlock, *dbp);
+                        }
```

Comment:
You could test either variable here (probably `blocks_with_unknown_parent` would be best), but I assume you're testing both since the code references both. The only downside is it could lead the casual reader to think that only one of these could be `nullptr` and not the other. The `emplace` simplification is nice here.

Reply:
> ... but I assume you're testing both since the code references both.

Correct.